### PR TITLE
Add contribution guide and code of conduct

### PR DESCRIPTION
**Pull request by dondonz**
This PR
* Adds a contribution guide and explains the PR review process. This website information is mirrored in the CONTRIBUTING.md file in case a contributor finds the GitHub repo first
* Moves contribution information from the implementation guide into a separate document
* Adds explanation on immutable specifications, except for minor non-semantic changes. Add versioning explanation.
* Adds a code of conduct (copied from graphql-java)
* Adds explanation of the Contributor License Agreement (copied from graphql-wg)

After merging this PR, the remaining two items are: deciding on license and connecting to the GraphQL Foundation domain.

### CODE REVIEW

File: scalars/guide.md [hunk 1]
```diff
@@ -166,30 +166,3 @@ values for logically identical values. For example a `MyLocalDate` scalar could
 accept the literals `"01-10-2022"` and `"01102022"` as input for the first of
 October 2022, but the result coercion should always return one of the possible
 representations.
-
```
Comment (dondonz) [inline scalars/guide.md:169]: Moved to template

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# Contributor Code of Conduct
```
Comment (andimarek) [inline CODE_OF_CONDUCT.md:1]: I think we should reference here the general GraphQL code of conduct and not introduce a new one.

File: scalars/contributed/andimarek/date-time.md [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 "Author: andimarek"
 
-"Date: 18.10.2022"
+"Date: 2022.10.18"
```
Comment (andimarek) [inline scalars/contributed/andimarek/date-time.md:5]: How about yyyy-mm-dd instead of `.`?

File: scalars/new-scalar.md [hunk 4]
```diff
@@ -1,6 +1,71 @@
...
+# Custom Scalar specification contribution guide
 
-In order to contribute a new Scalar, please open a new PR for each new Scalar.
+Thanks for contributing to GraphQL Scalars.
 
-Copy either "template.md" or "template-string.md" and modify it. The new file
-should be under "scalars/contributed/<your-name>/<scalar-name>.md"
+The goal of the GraphQL Scalars project is to provide a directory of GraphQL
+Custom Scalar specifications, contributed by the community. Contributed
+specifications will be hosted on a GraphQL Foundation owned domain, which can be
```
Comment (andimarek) [inline scalars/new-scalar.md:7]: We should be specific and name the domain scalars.graphql.org

File: scalars/new-scalar.md [hunk 5]
```diff
@@ -1,6 +1,71 @@
...
+correctness of the specification is the responsibility of the original
+contributor.
+
+## Immutable specifications
+
+Specification semantics must not change, as specifications are publicly
+available reference documents. We will permit small edits which do not change
+specification semantics, such as typo fixes.
+
+A new version of a Custom Scalar specification should have a new scalar name,
```
Comment (andimarek) [inline scalars/new-scalar.md:70]: or just a completely different name like `OffsetDateTime` vs `DateTime`

File: CODE_OF_CONDUCT.md (hunk 2 above)
Comment (dondonz) [reply CODE_OF_CONDUCT.md:1]: Good idea, I'll reference https://graphql.org/codeofconduct/

**Pull request merged**